alf/algorithms/ddpg_algorithm.py

```diff
@@ -140,7 +141,8 @@ def __init__(self,
         """
         self._calculate_priority = calculate_priority
         if epsilon_greedy is None:
-            epsilon_greedy = config.epsilon_greedy if config else 0.
+            epsilon_greedy = alf.get_config_value(
+                'TrainerConfig.epsilon_greedy', override=config.epsilon_greedy)
```

Comment:
When config is None, this still throws an error of attribute 'epsilon_greedy' error.

Reply:
You are right.  This was caught by unit test and fixed.

Reply:
> You are right. This was caught by unit test and fixed.

I might not be clear about the first comment. I think the problem with this is that when a user use either SAC or DDPG as a child algorithm (it can be nested anywhere under any algorithm) and forgets to pass the `config` object to it, then accessing `epsilon_greedy` in this child algorithm will report an error.